dask_geopandas/core.py

```diff
@@ -306,6 +307,30 @@ def explode(self):
     def cx(self):
         return _CoordinateIndexer(self)
 
+    def hilbert_distance(self, p=15):
+
+        """
+        A function that calculates hilbert distance for each geometry
+        in each partition of a Dask-GeoDataFrame
+
+        Parameters
+        ----------
+        p	: Hilbert curve parameter
+
+        Returns
+        ----------
+        Distances for each partition
+        """
+
+        # Compute total bounds of all partitions rather than each partition
+        total_bounds = self.total_bounds.compute()
+        # Calculate hilbert distances for each partition
+        distances = self.map_partitions(
+            lambda s: _hilbert_distance(s, total_bounds=total_bounds, p=p)
```

Comment:
```suggestion
            lambda s, total_bounds: _hilbert_distance(s, total_bounds, p=p), total_bounds
```

Untested, but something like this ;)

Reply:
This indeed works more or less like that. See the docs for `map_partitions`: https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.Series.map_partitions

If we are not directly passing the arguments to the function, we also don't need a lambda:

```python
        distances = self.map_partitions(
            _hilbert_distance, total_bounds=total_bounds, p=p,
        )
```

Quickly tried the above, and then it gives a problem with the meta inference (apparently for inferring the result dtype based on the dask dataframe "meta" object, it does pass the delayed total_bounds, which then doesn't work). So we will need to manually specify the result dtype (and can then also directly specify a name for the Series): `            meta=("hilbert_distance", "int64")` passed as extra argument to `map_partitions`.


I tried those fixes with a small example, and visualized the graph:

```
df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))
ddf = dask_geopandas.from_geopandas(df, npartitions=4)
d = ddf.geometry.hilbert_distance()
d.visualize()
```

![image](https://user-images.githubusercontent.com/1020496/125041237-d1890200-e098-11eb-9baf-86c692a1a028.png)

where you can see nicely that it first computes the total_bounds from all partitions, and passing that to the hilbert_distance task (so in practice this will require loading the data twice if they don't all fit into memory, but so not now at least those two steps are fully lazy in the task graph instead of calling compute() for the total_bounds up front)

Reply:
Nice! I have added this to 0fba9a8